ssl.go

```diff
@@ -0,0 +1,110 @@
+//go:build !go_tarantool_ssl_disable
+// +build !go_tarantool_ssl_disable
+
+package tarantool
+
+import (
+	"errors"
+	"io/ioutil"
+	"net"
+	"time"
+
+	"github.com/tarantool/go-openssl"
+)
+
+func sslDialTimeout(network, address string, timeout time.Duration,
+	opts SslOpts) (connection net.Conn, err error) {
+	var ctx interface{}
+	if ctx, err = sslCreateContext(opts); err != nil {
+		return
+	}
+
+	return openssl.DialTimeout(network, address, timeout, ctx.(*openssl.Ctx), 0)
+}
+
+// interface{} is a hack. It helps to avoid dependency of go-openssl in build
+// of tests with the tag 'go_tarantool_ssl_disable'.
+func sslCreateContext(opts SslOpts) (ctx interface{}, err error) {
+	var sslCtx *openssl.Ctx
+
+	// Require TLSv1.2, because other protocol versions don't seem to
+	// support the GOST cipher.
+	if sslCtx, err = openssl.NewCtxWithVersion(openssl.TLSv1_2); err != nil {
+		return nil, err
+	}
+	ctx = sslCtx
+	sslCtx.SetMaxProtoVersion(openssl.TLS1_2_VERSION)
+	sslCtx.SetMinProtoVersion(openssl.TLS1_2_VERSION)
+
+	if opts.CertFile != "" {
+		if err = sslLoadCert(sslCtx, opts.CertFile); err != nil {
+			return
```

Comment:
Is there any difference between `return nil, err` on line 33 and `return` here and below? They seem identical to me consider the code,

Reply:
Thank you, fixed.